apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts

```diff
@@ -1,18 +1,33 @@
 import { SaleorSyncWebhook } from "@saleor/app-sdk/handlers/next";
-import { UntypedCalculateTaxesDocument } from "../../../../generated/graphql";
+import {
+  CalculateTaxesEventFragment,
+  UntypedCalculateTaxesDocument,
+} from "../../../../generated/graphql";
 import { saleorApp } from "../../../../saleor-app";
 import { createLogger } from "../../../lib/logger";
-import { calculateTaxesPayloadSchema, ExpectedWebhookPayload } from "../../../lib/saleor/schema";
-import { getAppConfig } from "../../../modules/app-configuration/get-app-config";
-import { ActiveTaxProvider } from "../../../modules/taxes/active-tax-provider";
+import { getActiveTaxProvider } from "../../../modules/taxes/active-tax-provider";
 
 export const config = {
   api: {
     bodyParser: false,
   },
 };
 
-export const orderCalculateTaxesSyncWebhook = new SaleorSyncWebhook<ExpectedWebhookPayload>({
+type CalculateTaxesPayload = Extract<CalculateTaxesEventFragment, { __typename: "CalculateTaxes" }>;
+
+function verifyCalculateTaxesPayload(payload: CalculateTaxesPayload) {
+  if (!payload.taxBase.lines) {
+    throw new Error("No lines found in taxBase");
+  }
+
+  if (!payload.taxBase.address) {
+    throw new Error("No address found in taxBase");
+  }
+
+  return payload;
+}
```

Comment:
I think its reused code, maybe you can extract some shared logic

Reply:
I think it makes sense to be a bit more redundant here. The logic is the same for both webhooks, since the payload is the same, but it doesn't have to be. It's an important part of flow, I like keeping it next to the webhook handler. If it grows and is still the same, I can extract it